openssl-dynamic/src/main/c/tcn_atomic.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#include "tcn_atomic.h"
+#include <atomic>
+
+tcn_atomic_uint32_t tcn_atomic_uint32_new() {
+    auto *atomic = new std::atomic<uint32_t>(0);
+    return (tcn_atomic_uint32_t) atomic;
+}
+
+void tcn_atomic_uint32_free(tcn_atomic_uint32_t atomic) {
+    auto *p = (std::atomic<uint32_t> *) atomic;
```

Comment:
consider removing the temp variable.

also consider passing a `**` as a parameter so you can always set `*atomic = NULL;` in these methods.